go/app/main.go

```diff
@@ -121,36 +126,55 @@ func addItem(c echo.Context) error {
 	}
 
 	//Insert the data into the database
-	statement, _ := database.Prepare("INSERT INTO `Items` (`id`, `name`, `category_id`, `image_filename`) VALUES (?, ?, ?, ?);")
+	statement, err := database.Prepare("INSERT INTO `Items` (`name`, `category_id`, `image_filename`) VALUES (?, ?, ?);")
 	if err != nil {
 		log.Fatal(err)
 	}
 	defer statement.Close()
 
-	//Getting the id corresponding to the last item
-	var itemID int
-	err = database.QueryRow("SELECT id FROM Items ORDER BY id DESC LIMIT 1").Scan(&itemID)
-	if err != nil {
-		log.Fatal(err)
-	}
-
 	//Getting the id corresponding to the category that was given
 	var categoryID int
 	err = database.QueryRow("SELECT id FROM Category WHERE name = ?", category).Scan(&categoryID)
 	if err != nil {
 		fmt.Println("This category does not exist")
-		log.Fatal(err)
+		addCategory(category)
+		err = database.QueryRow("SELECT id FROM Category WHERE name = ?", category).Scan(&categoryID)
+		if err != nil {
+			log.Fatal(err)
+		}
 	}
 
 	//Execute the INSERT statement with the values
-	_, err = statement.Exec((itemID + 1), name, categoryID, newImageName)
+	_, err = statement.Exec(name, categoryID, newImageName)
 	if err != nil {
 		log.Fatal(err)
 	}
 
 	return c.JSON(http.StatusOK, res)
 }
 
+//addCategory is called when there is no Category when creating a new item
+func addCategory(category string) {
+	prepareDB()
+	database, err := sql.Open("sqlite3", "mercari.db")
```

Comment:
I overlooked database is not closed. Could you close databases like statements?
https://pkg.go.dev/database/sql#DB.Close